molecularnodes/annotations/manager.py

```diff
@@ -0,0 +1,315 @@
+        return [i.name for i in self._interfaces.values()]
+
+    @classmethod
+    def register(cls, annotation_class) -> None:
+        """
+        Register an annotation class
+
+        This method adds the annotation class to the entity specific class
+        registry (_classes) and adds a new method (add_<annotation_type>) to
+        the manager with a signature that matches the annotation inputs.
+
+        """
+        cls._validate_annotation_class(annotation_class)
+        # Add to Entity class specific registry
+        cls._classes[annotation_class.annotation_type] = annotation_class
+        # Add method to Entity specific manager
+        method_name = f"add_{annotation_class.annotation_type}"
+        method = functools.partialmethod(
+            cls._create_annotation_instance, annotation_class
+        )
+        parameters = [
+            inspect.Parameter("self", inspect.Parameter.POSITIONAL_ONLY),
+            inspect.Parameter("annotation_class", inspect.Parameter.POSITIONAL_ONLY),
+        ]
+        # Add the annotation inputs as keyword params
+        py_annotations = get_all_class_annotations(annotation_class)
+        for key, value in py_annotations.items():
+            default = inspect.Parameter.empty
+            # set the default if the param in annotation class is a class
+            # attribute and not just an annotation
+            if hasattr(annotation_class, key):
+                default = getattr(annotation_class, key)
+            param = inspect.Parameter(
+                key, inspect.Parameter.KEYWORD_ONLY, annotation=value, default=default
+            )
+            parameters.append(param)
+        # Create method signature to match annotation class inputs
+        method.func.__signature__ = inspect.Signature(parameters)
+        # Update annotation properties
+        cls._update_annotation_props(annotation_class)
+        setattr(cls, method_name, method)
+
+    @classmethod
+    def _update_annotation_props(cls, annotation_class: BaseAnnotation):
+        """Update annotation properties attached to Object"""
+        AnnotationProperties = type(
+            "AnnotationProperties", (BaseAnnotationProperties,), {}
+        )
+        # Add each annotation type inputs as a pointer to a separate property group
+        for annotation_class in cls._classes.values():
+            annotation_type = annotation_class.annotation_type
+            AnnotationInputs = create_annotation_type_inputs(annotation_class)
+            bpy.utils.register_class(AnnotationInputs)
+            AnnotationProperties.__annotations__[annotation_type] = (
+                bpy.props.PointerProperty(type=AnnotationInputs)
+            )
+        # Re-register the new AnnotationProperties class
+        bpy.utils.register_class(AnnotationProperties)
+        # Re-assign the annotation properties to Object - old data is retained
+        bpy.types.Object.mn_annotations = bpy.props.CollectionProperty(
```

Comment:
I'm a bit unsure about this. As this is working on the `bpy.types.Object`, this changes it for every object in the scene all at once? Every time this is invoked this would remove all existing collection properties?